app/src/main/java/org/gdg/frisbee/android/pulse/PulseAdapter.java

```diff
@@ -117,23 +123,16 @@ public void setPulse(final int mode, Pulse pulse) {
         mMode = mode;
         mPulse.clear();
         mPulse.addAll(pulse.entrySet());
-        mPosition = new int[mPulse.size()];
+        //Only initialize if the positions are not provided.
+        //They are provided in constructor and coming from savedInstanceState of the Fragment.
+        if (mPositions == null || mPositions.length != mPulse.size()) {
```

Comment:
We don't know the size before. Network operation is done, Pulse object is filled and then this is called. That's where we know the size.